### PR TITLE
Block explorer privacy warnings

### PR DESCRIPTION
**Pull request by michaelWuensch**
## Description

This PR adds a privacy warning that appears when looking up a transaction or address with a blockexplorer.
It also adds the posibility to look up the funding transaction in the chanel detail view with a block explorer.

## Motivation and Context

Users should now which risks they take.

## How Has This Been Tested?

On my Samsung S9, API 29

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [Contribution document](../docs/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: app/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -175,6 +175,7 @@
     <string name="guardian_tooMuchMoney">Congratulations, you seem to have a nice stack of Bitcoin!\nPlease keep in mind that Zap is a Hot Wallet.\nWe do our best to keep this wallet secure, but for your own security sake, we recommend you consider Cold Wallets or Hardware Wallets to store your major savings.</string>
     <string name="guardian_notReadyForMainnet">You are on MAINNET with real Bitcoin!\nZap Android is in early development phase. You might lose funds!\n\nIf you don\'t want to take that risk, switch to testnet in the meantime.</string>
     <string name="guardian_remoteConnect">Are you sure you want to connect to \n\n%1$s\n\nPlease check the hostname carefully.</string>
+    <string name="guardian_blockExplorer">By looking up information with a Block Explorer your IP-Address gets exposed. In most cases this is no problem, but if you want to stay 100%% anonymous, you should either not use it or go to the settings and choose a Block Explorer that supports tor.</string>
```
Comment (maraph) [inline app/src/main/res/values/strings.xml]: typo: `100%%` - maybe rephrase like `if you want to increase your anonymity`.  Would not suggest the user that there is 100% anonymity possible.  Would capitalize **T**or

File: app/src/main/java/zapsolutions/zap/util/BlockExplorer.java [hunk 2]
```diff
@@ -6,123 +6,143 @@
...
      * Shows address details in a browser window using the preferred block explorer.
      *
      * @param address address to show
      * @param ctx
      */
-    public static void showAddress(String address, Context ctx) {
+    public void showAddress(String address, Context ctx) {
+        mContext = ctx;
         String explorer = PrefsUtil.getPrefs().getString("blockExplorer", "BlockCypher");
         boolean mainnet = !Wallet.getInstance().isTestnet();
```
Comment (maraph) [inline app/src/main/java/zapsolutions/zap/util/BlockExplorer.java]: Would prefer `isMainnet`

File: app/src/main/java/zapsolutions/zap/util/BlockExplorer.java [hunk 3]
```diff
@@ -6,123 +6,143 @@
...
 
 /**
  * This class allows showing details of On-Chain transactions and addresses using
  * 3rd Party services (Block Explorers)
  */
-public class BlockExplorer {
+public class BlockExplorer implements UserGuardianInterface {
 
+    private String mUrl;
+    private boolean mSupported;
```
Comment (maraph) [inline app/src/main/java/zapsolutions/zap/util/BlockExplorer.java]: `mIsNetworkSupported` ?

File: app/src/main/java/zapsolutions/zap/util/BlockExplorer.java [hunk 4]
```diff
@@ -6,123 +6,143 @@
...
+    private Context mContext;
+
+
+    private void unsupportedNetwork(String explorer, String network, Context ctx) {
```
Comment (maraph) [inline app/src/main/java/zapsolutions/zap/util/BlockExplorer.java]: I would move the private methods of the class to the bottom and the public to the top 👍

File: app/src/main/java/zapsolutions/zap/util/BlockExplorer.java [hunk 5]
```diff
@@ -6,123 +6,143 @@
...
 
-        if (supported) {
-            // Call the url
-            Intent browserIntent = new Intent(Intent.ACTION_VIEW, Uri.parse(url));
-            ctx.startActivity(browserIntent);
+        if (explorer.equals("Blockstream (v3 Tor)")){
+            startBlockExplorer();
         } else {
-            unsupportedNetwork(explorer, mainnet ? "mainnet" : "testnet", ctx);
+            new UserGuardian(mContext, this).privacyBlockExplorer();
```
Comment (maraph) [inline app/src/main/java/zapsolutions/zap/util/BlockExplorer.java]: Just an idea, maybe for another PR.
If we would pass a `Bundle` with extras to the `UserGuardian` class, we could pass the url like
bundle.putExtras(BlockExplorerURL,url)
and receive it in `guardianDialogConfirmed(String DialogName, Bundle extras)` again.
We then could remove the `DialogName` as it could part of the `extras` like
`bundle.putExtras(DialogType,"blockexplorer")`

This is just pseudecode to translate the idea 👍 
Let me know what you think. We could create an issue and tackle it later as well.

File: app/src/main/res/values/strings.xml (hunk 1 above)
Comment (michaelWuensch) [reply app/src/main/res/values/strings.xml]: 100%% is actually correct, if you only have 100% it will cause a lint error. % is escaping
But I like your suggestion, so it will be removed anyway.

File: app/src/main/java/zapsolutions/zap/util/BlockExplorer.java (hunk 5 above)
Comment (michaelWuensch) [reply app/src/main/java/zapsolutions/zap/util/BlockExplorer.java]: I actually tried that first. I had the BlockExplorer Class as a static class with static functions.
I was able to transmit the URL with a bundle. But I also needed a Context to start the url from which cannot be included in the bundle. So I ended up with this solution where I thought that is easier.
